src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -859,6 +859,39 @@ private bool CheckForAssignmentToSelf(BoundAssignmentOperator node)
             return false;
         }
 
+        private void CheckForDeconstructionAssignmentToSelf(BoundTupleLiteral leftTuple, BoundExpression right)
+        {
+            while (right.Kind == BoundKind.Conversion)
+            {
+                right = ((BoundConversion)right).Operand;
```

Comment:
>right = ((BoundConversion)right).Operand; [](start = 16, length = 41)

I am somewhat uncomfortable about ignoring all possible conversions here. Would it make sense to limit them to standard tuple conversions, for example? What if conversion is user-defined?